text/0000-rtsan-and-attributes.md

```diff
@@ -0,0 +1,273 @@
+- Feature Name: RealtimeSanitizer, `nonblocking` and `blocking attributes
+- Start Date: 2025-01-30
+- RFC PR: 
+- Rust Issue: 
+
+# Summary
+[summary]: #summary
+
+Many software projects that utilize Rust are subject to real-time constraints. Software that is written for use in audio, embedded, robotics, and aerospace must adhere to strict deterministic-time execution or face consequences that may be catastrophic. LLVM 20 introduces RealtimeSanitizer one approach to detecting and reporting disallowed non-deterministic execution time calls in real-time contexts.
+
+This RFC proposes that RealtimeSanitizer be integrated into the Rust ecosystem. To serve that end, we propose a few changes, outlined in this document:
+
+1. RealtimeSanitizer can be enabled in unstable mode - like the other sanitizers
+2. The introduction of `nonblocking` (marking a function as real-time constrained) and `blocking` (marking a function as inappropriate for use in a `nonblocking` context)
+3. The addition of the `rtsan_scoped_disabler!` macro
+4. Disabling rtsan for the `panic!` and `assert*!` macros
+
+# Motivation
+[motivation]: #motivation
+ 
+Increasingly, Rust is being used in problem spaces that are real-time constrained, such as audio, robotics, aerospace and embedded. Real-time programming is defined by deadlines - if a solution is not provided by a specific deadline, some consequence may occur. 
+
+For example:
+
+> In an autonomous vehicle perception subsystem, it is not enough to detect an obstacle and decide to stop in some unknown amount of time. You must detect the obstacle AND stop within N ms, or you may crash.
+
+> In audio, you must fill a buffer and pass it back to the operating system within N ms, otherwise your user may hear a click or pop which may damage their audio equipment, or minimally annoy them.
+
+> In aerospace guidance systems if your software doesn't update on a regular tick your simulation of what is happening may diverge from reality. Unfortunately this may also mean your rocket converges with the ground.
+
+Code in these environments must run in a deterministic amount of time. Allocations, locks, and other OS resource access are disallowed because they don't have bound upper execution time.
+
+**Historically, it has been very difficult for programmers to detect, RealtimeSanitizer is one approach to detecting these issues before they run on end-users machines.**
+
+A few resources that go into more depth on real-time programming:
+* https://en.wikipedia.org/wiki/Real-time_computing
+* http://www.rossbencina.com/code/real-time-audio-programming-101-time-waits-for-nothing
+* https://www.youtube.com/watch?v=ndeN983j_GQ
+
+# Guide-level explanation
```

Comment:
This section was tweaked to sound like I am describing "the existing Rust implementation" as recommended in the template. i.e. I'm describing to a new user how to user RTSan in rust as if it exists today